### DISCUSSION TITLE
v2.0.0-alpha.1: Release 2.0.0-alpha.1 (#1121)

### DISCUSSION THREAD

**Discussion by srikrsna-buf** [connectrpc/connect-es#1131] [Announcements]
## What's new in version 2

To support protobuf editions, `@bufbuild/protobuf` had to make breaking changes, more on this [here](https://github.com/bufbuild/protobuf-es/pull/801). Upgrading to `v2` of `@bufbuild/protobuf` will be breaking change for connect users. 

The most notable change is that v2 doesn't require a separate plugin anymore! we only need `protoc-gen-es`. For most users this will be a simple change of just removing the connect plugin and changing the import path to point to the protobuf generated types:

```ts
import { createPromiseClient } from "@connectrpc/connect";
import { createConnectTransport } from "@connectrpc/connect-node";
// Before this was import { ElizaService } from "./gen/eliza_connect.js"
import { ElizaService } from "./gen/eliza_pb.js";

// Alternatively, use createGrpcTransport or createGrpcWebTransport here
// to use one of the other supported protocols.
const transport = createConnectTransport({
  httpVersion: "2",
  baseUrl: "https://localhost:8443",
  nodeOptions: { rejectUnauthorized },
});

const client = createPromiseClient(ElizaService, transport);
const res = await client.say({ sentence });
```

Please note that this is an alpha release, and APIs might still change. We're also missing documentation yet.  But if you want to try it out, we welcome your feedback!

This release is published with the `alpha` tag. To install the alpha packages, you can run:

```bash
npm install @connectrpc/connect@alpha @bufbuild/protobuf@beta @bufbuild/protoc-gen-es@beta
```

<hr /><em>This discussion was created from the release <a href='https://github.com/connectrpc/connect-es/releases/tag/v2.0.0-alpha.1'>v2.0.0-alpha.1: Release 2.0.0-alpha.1 (#1121)</a>.</em>

**Comment by biziosan** [comment]
Hi, great improvement!

There seems to be a bug in this new version, though. `Any` does not serialize. I wrote a small example to reproduce the issue, and I can file an issue if you want. The same code and basic types work just fine in `@bufbuild/protobuf`.

**Reply by timostamm** [reply]
Apologies for the late reply. Please file an issue and we'll take a look!

**Comment by nicole0707** [comment]
Just wondering the estimated time for the release of the v2, Thanks!

**Reply by timostamm** [reply]
I'm sorry it hasn't been released yet 😞 

We found a couple of regressions to fix and some new issues to solve before we can cut a stable version. We've made good progress in the last couple of weeks. I don't have a fixed date unfortunately, but we're committed to get it out the door as soon as possible. I guess the fixes will take two more weeks, and then some time to update the documentation.

**Reply by nicole0707** [reply]
> I'm sorry it hasn't been released yet 😞
> 
> We found a couple of regressions to fix and some new issues to solve before we can cut a stable version. We've made good progress in the last couple of weeks. I don't have a fixed date unfortunately, but we're committed to get it out the door as soon as possible. I guess the fixes will take two more weeks, and then some time to update the documentation.

Thanks @timostamm! Amazing work! We are planning to kick off the upgrade to v2 alpha next week and will keep you updated if any issues arise!

**Reply by timostamm** [reply]
Awesome! We just cut a [beta version](https://github.com/connectrpc/connect-es/releases/tag/v2.0.0-beta.1) and don't plan to make any API changes. It should also be much easier to give it a try with the migration tool.
